src/external_registration.cpp

```diff
@@ -37,17 +46,20 @@ int main(int argc, char** argv)
   ros::removeROSArgs( argc, argv, args_out );
 
   namespace po = boost::program_options;
-  po::options_description desc("Options");
+  po::options_description desc("Options"), all_desc("Options");
   desc.add_options()
     ("help,h", "print help message")
     ("roscore_ip,r", po::value<std::string>(), "set the ip of the roscore to use")
-    ("network_interface,i", po::value<std::string>()->default_value("eth0"),  "set the network interface over which to connect")
-    ("namespace,n", po::value<std::string>()->default_value(""), "set an explicit namespace in case ROS namespace variables cannot be used");
+    ("network_interface,i", po::value<std::string>()->default_value("eth0"),  "set the network interface over which to connect");
+
+  all_desc.add_options()
+    ("node_name,n", po::value<std::string>()->default_value("naoqi_driver"), "this node name");
+  all_desc.add(desc);
 
   po::variables_map vm;
   try
   {
-    po::store( po::parse_command_line(argc, argv, desc), vm );
+    po::store( po::command_line_parser(argc, argv).options(all_desc).extra_parser(ros_node_name_parser).run(), vm );
```

Comment:
@vrabaud We rather need to use `__name` than remove it to pass this value to `naoqi::Driver`.